### PR TITLE
Fix DisposeTrack not working as expected

### PR DESCRIPTION
**Pull request by peppy**
Regression caused by the new async logic. Disposed tracks would be returned after the first dispose.

### CODE REVIEW

File: osu.Game/Beatmaps/WorkingBeatmap.cs [hunk 1]
```diff
@@ -114,11 +115,57 @@ private void applyRateAdjustments(Track t = null)
...
+                    return lazy.IsValueCreated;
+                }
+            }
+
+            public Task<T> Value
+            {
+                get
+                {
+                    ensureValid();
+                    return lazy.Value;
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/WorkingBeatmap.cs]: You realise that doing this removes any thread-safety guarantees Lazy.Value previously provided, right?

Comment (peppy) [reply osu.Game/Beatmaps/WorkingBeatmap.cs]: how so?

Comment (smoogipoo) [reply osu.Game/Beatmaps/WorkingBeatmap.cs]: If two threads enter Value -> init() at the same time, then one thread continues on to return lazy.Value, and the other thread does the same afterwards, the object will be initialised twice.

Comment (peppy) [reply osu.Game/Beatmaps/WorkingBeatmap.cs]: fair call. will add some local locking.

**Review by smoogipoo — APPROVED** [review]
This looks ok to me

**Pull request merged**